### PR TITLE
add GET fields endpoint and test cases for related viewset

### PR DESCRIPTION
**Pull request by Tayef-Shah**
### Notion ticket link

[GET Fields Endpoint](https://www.notion.so/uwblueprintexecs/GET-fields-endpoint-c853ef7d681548b7862233a87f650f96)

Quick note: I didn't make a test file for  the serializer since it's a simple serializer

### Implementation description

- added a serializer
- added a viewset
- registered the endpoint with the router
- added test cases for the viewset

### Steps to test

1. Run tests on registration app

### What should reviewers focus on?

- Functionality & readability

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: registration/views.py [hunk 1]
```diff
@@ -22,3 +22,17 @@ def get_serializer_class(self):
         if self.action in ["retrieve", "create"]:
             return FamilyDetailSerializer
         return FamilySerializer
+
+
+class FieldViewSet(
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.CreateModelMixin,
```
Comment (brittanylau) [inline registration/views.py]: we might not need these two for now - can we remove em and update the tests accordingly?

File: registration/serializers.py [hunk 2]
```diff
@@ -97,3 +97,16 @@ def create(self, validated_data):
             family.save()
 
         return family
+
+
+class FieldSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = Field
+        fields = [
```
Comment (brittanylau) [inline registration/serializers.py:105]: can we also include the id?

**Review by brittanylau — APPROVED** [review]
thanks for the speedy turnaround! 💨 

just a couple quick notes, feel free to merge after addressing

File: registration/views.py (hunk 1 above)
Comment (Tayef-Shah) [reply registration/views.py]: ok gotcha, just listing out all the fields is good enough for now

**Pull request merged**